libsolidity/interface/Natspec.cpp

```diff
@@ -99,16 +99,46 @@ Json::Value Natspec::devDocumentation(ContractDefinition const& _contractDef)
 		if (auto fun = dynamic_cast<FunctionDefinition const*>(&it.second->declaration()))
 		{
 			Json::Value method(devDocumentation(fun->annotation().docTags));
-			if (!method.empty())
-				// add the function, only if we have any documentation to add
+
+			// add the function, only if we have any documentation to add
+			if (!method.empty()) {
+				Json::Value ret(Json::objectValue);
+
+				// for constructors, the "return" node will never exist. invalid tags
+				// will already generate an error within dev::solidity::DocStringAnalyzer.
+				auto const& returnParams = fun->returnParameters();
+				auto returnDoc = fun->annotation().docTags.equal_range("return");
+
+       			// if there is no name then append subsequent return notices like previous behavior
+       			string value;
+				unsigned int n = 0;
+				for (auto i = returnDoc.first; i != returnDoc.second; i++) {
+          			string paramName = returnParams[n]->name();
+
+					if (!paramName.empty()) {
+						ret[paramName] = Json::Value(i->second.content);
+						++n;
+					}
+
+					else {
+            			value += i->second.content;
```

Comment:
Right now this piece of code handles multiple **unnamed** return parameters by concatenating all the `return` doc notices it encounters like before.